File: app/src/main/java/com/commit451/internallinkmovementmethod/sample/MainActivity.java
package com.commit451.internallinkmovementmethod.sample;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Html;
import android.widget.TextView;
import android.widget.Toast;

import com.commit451.internallinkmovementmethod.InternalLinkMovementMethod;

public class MainActivity extends AppCompatActivity {

    private static final String TEXT = "Featured on <a href=\"https://android-arsenal.com/details/1/1762\">Android Arsenal</a>";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView textView = findViewById(R.id.text);
        textView.setText(Html.fromHtml(TEXT));
        textView.setMovementMethod(new InternalLinkMovementMethod() {
            @Override
            protected void onLinkClicked(@NonNull TextView textView, @NonNull String link, @Nullable String text) {
                Toast.makeText(MainActivity.this, "Link: " + link + " with text: " + text + " clicked", Toast.LENGTH_LONG)
                        .show();
            }
        });
    }
}

File: app/src/main/java/com/commit451/internallinkmovementmethod/sample/MainActivity.kt
package com.commit451.internallinkmovementmethod.sample

import android.os.Bundle
import android.text.Html
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

import com.commit451.internallinkmovementmethod.InternalLinkMovementMethod

class MainActivity : AppCompatActivity() {

    companion object {
        private const val TEXT = "Featured on <a href=\"https://android-arsenal.com/details/1/1762\">Android Arsenal</a>"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val textView = findViewById<TextView>(R.id.text)
        textView.text = Html.fromHtml(TEXT)
        textView.movementMethod = object : InternalLinkMovementMethod() {
            override fun onLinkClicked(textView: TextView, link: String, text: String?) {
                Toast.makeText(this@MainActivity, "Link: $link with text: $text clicked", Toast.LENGTH_LONG)
                        .show()
            }
        }
    }
}
